server.py

```diff
@@ -236,9 +245,16 @@ def question(code, number):
         .dicts()
         .get()
     )
-    return bottle.template(
-        "question.html", question_number=number, contest=code, question=statement
-    )
+    user = Session.get(Session.token == bottle.request.get_cookie("s_id")).user
+    question_status = QuestionStatus.select().where((QuestionStatus.question == number) & (QuestionStatus.user == user))
+    if question_status.count() == 0:
+        return bottle.template(
+            "question.html", question_number=number, contest=code, question=statement, question_status='Not Submitted'
+        )
+    else:
+        return bottle.template(
+            "question.html", question_number=number, contest=code, question=statement, question_status='Submitted'
+        )
 
```

Comment:
It would be better to directly return the `question_status` extracted from the query. Also, right now the status shows Submitted/ Not Submitted, it would be better to instead use Accepted/ Wrong Answer/ Unattempted.

Reply:
We can also avoid the if-else block by using something like:
```
question_status = QuestionStatus.select(QuestionStatus.status).where(
        (QuestionStatus.question == number) & (QuestionStatus.user == user)
    ).dicts().get()
return bottle.template(
    "question.html", question_number=number, contest=code, question=statement,
    question_status= ((question_status[status]==Null)?"Unattempted":question_status[status])
)
```